File: C++/easy/space-age/space_age.cpp
#include "space_age.h"

namespace space_age {
  space_age::space_age(const long int& seconds): seconds_(seconds), kSecondsInAEarthYear_(31557600) {}

  int space_age::seconds(void) const { return seconds_; }

  double space_age::on_mercury(void) const { return double(seconds_) / (kSecondsInAEarthYear_ * 0.2408467); }

  double space_age::on_venus(void) const { return double(seconds_) / (kSecondsInAEarthYear_ * 0.61519726); }

  double space_age::on_earth(void) const { return double(seconds_) / kSecondsInAEarthYear_; }

  double space_age::on_mars(void) const { return double(seconds_) / (kSecondsInAEarthYear_ * 1.8808158); }

  double space_age::on_jupiter(void) const { return double(seconds_) / (kSecondsInAEarthYear_ * 11.862615); }

  double space_age::on_saturn(void) const { return double(seconds_) / (kSecondsInAEarthYear_ * 29.447498); }

  double space_age::on_uranus(void) const { return double(seconds_) / (kSecondsInAEarthYear_ * 84.016846); }

  double space_age::on_neptune(void) const { return double(seconds_) / (kSecondsInAEarthYear_ * 164.79132); }

}  // namespace space_age

File: Python/easy/space-age/space_age.py
from math import floor

class SpaceAge:
    def __init__(self, seconds):
        self.__seconds = seconds

    def on_mercury(self):
        return round(self.on_earth() / 0.2408467, 2)

    def on_venus(self):
        return floor((self.on_earth() / 0.61519726) * 100) / 100
    
    def on_earth(self):
        return round(self.__seconds / (365.25 * 24 * 3600), 2)

    def on_mars(self):
        return round(self.on_earth() / 1.8808158, 2)

    def on_jupiter(self):
        return round(self.on_earth() / 11.862615, 2)
    
    def on_saturn(self):
        return round(self.on_earth() / 29.447498, 2)

    def on_uranus(self):
        return round(self.on_earth() / 84.016846, 2)

    def on_neptune(self):
        return round(self.on_earth() / 164.79132, 2)
